src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -135,30 +135,10 @@ internal static byte[][] SegmentedEncodeUnsignedInteger(uint value)
         /// </summary>
         /// <param name="bigEndianBytes">The value to encode, in big integer representation.</param>
         /// <returns>The encoded segments { tag, length, value }</returns>
-        internal static byte[][] SegmentedEncodeUnsignedInteger(byte[] bigEndianBytes)
+        internal static byte[][] SegmentedEncodeUnsignedInteger(ReadOnlySpan<byte> bigEndianBytes)
         {
-            Debug.Assert(bigEndianBytes != null);
-
-            return SegmentedEncodeUnsignedInteger(bigEndianBytes, 0, bigEndianBytes.Length);
-        }
-
-        /// <summary>
-        /// Encode the segments { tag, length, value } of an unsigned integer represented within a bounded array.
-        /// </summary>
-        /// <param name="bigEndianBytes">The value to encode, in big integer representation.</param>
-        /// <param name="offset">The offset into bigEndianBytes to read</param>
-        /// <param name="count">The count of bytes to read, must be greater than 0</param>
-        /// <returns>The encoded segments { tag, length, value }</returns>
-        internal static byte[][] SegmentedEncodeUnsignedInteger(byte[] bigEndianBytes, int offset, int count)
-        {
-            Debug.Assert(bigEndianBytes != null);
-            Debug.Assert(offset >= 0);
-            Debug.Assert(count > 0);
```

Comment:
Can you clarify?  Seems like it's actually stronger now than it was, in that if you give it invalid data that would have failed one of these asserts, it'll now cause an exception when constructing the span to pass to this.